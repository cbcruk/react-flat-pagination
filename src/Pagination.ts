import { PaginationProps } from './types'
import { usePagination } from './usePagination'

export function Pagination({
  totalNumber,
  pageSize,
  pageNumber = 1,
  children,
}: PaginationProps) {
  const result = usePagination({ totalNumber, pageSize, pageNumber })

  return children(result)
}
