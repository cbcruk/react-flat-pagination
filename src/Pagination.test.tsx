import { expect, test } from 'vitest'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Pagination } from './Pagination'

test('Pagination', async () => {
  const { container } = render(
    <Pagination total={100} pageCount={10} defaultCurrent={1}>
      {({
        current,
        ranges,
        handleNext,
        handlePrev,
        goTo,
        goToFirst,
        goToLast,
      }) => {
        return (
          <div>
            <div data-testid="current">{current}</div>
            <div>
              <button
                data-testid="first"
                onClick={() => {
                  goToFirst()
                }}
              >
                first
              </button>
              <button data-testid="prev" onClick={handlePrev}>
                prev
              </button>
              <div>
                {ranges.map((page) => {
                  return (
                    <a
                      key={page}
                      data-active={page === current}
                      data-testid={`item-${page}`}
                      onClick={() => {
                        goTo(page)
                      }}
                    >
                      {page}
                    </a>
                  )
                })}
              </div>
              <button data-testid="next" onClick={handleNext}>
                next
              </button>
              <button data-testid="last" onClick={goToLast}>
                last
              </button>
            </div>
          </div>
        )
      }}
    </Pagination>
  )
  expect(container).toMatchSnapshot()

  await userEvent.click(screen.getByTestId('next'))
  expect(screen.getByTestId('current')).toHaveTextContent('2')

  await userEvent.click(screen.getByTestId('first'))
  expect(screen.getByTestId('current')).toHaveTextContent('1')

  await userEvent.click(screen.getByTestId('item-2'))
  expect(screen.getByTestId('current')).toHaveTextContent('2')

  await userEvent.click(screen.getByTestId('last'))
  expect(screen.getByTestId('current')).toHaveTextContent('10')

  await userEvent.click(screen.getByTestId('first'))
  expect(screen.getByTestId('current')).toHaveTextContent('1')
})
