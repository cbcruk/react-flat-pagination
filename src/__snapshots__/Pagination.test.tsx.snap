// Vitest Snapshot v1

exports[`Pagination 1`] = `
<div>
  <div>
    <div
      data-testid="current"
    >
      1
    </div>
    <div>
      <button
        data-testid="first"
      >
        first
      </button>
      <button
        data-testid="prev"
      >
        prev
      </button>
      <div>
        <a
          data-active="true"
          data-testid="item-1"
        >
          1
        </a>
        <a
          data-active="false"
          data-testid="item-2"
        >
          2
        </a>
        <a
          data-active="false"
          data-testid="item-3"
        >
          3
        </a>
        <a
          data-active="false"
          data-testid="item-4"
        >
          4
        </a>
      </div>
      <button
        data-testid="next"
      >
        next
      </button>
      <button
        data-testid="last"
      >
        last
      </button>
    </div>
  </div>
</div>
`;
